# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/tests/browser/events/browser_test_scrolling.js
# Commit: 1e75ce1fef67
# Full Hash: 1e75ce1fef67f3139ac4f05993cad693c82c111c
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-02-17 09:50:03
# Regressor Bug: 1611060
# File Overlap Count: 1
# Description:
#   Bug 1611060 - Dispatch a11y scroll events for all DOM scrolls. r=Jamie
#   
#   Currently, only documents dispatch scroll events when in reality any
#   element can have scrollable overflows.
#   
# ==============================================================================

diff -r 20b59309d020 -r 1e75ce1fef67 accessible/tests/browser/events/browser_test_scrolling.js
--- a/accessible/tests/browser/events/browser_test_scrolling.js	Mon Feb 17 05:37:24 2020 +0200
+++ b/accessible/tests/browser/events/browser_test_scrolling.js	Mon Feb 17 04:09:34 2020 +0000
@@ -8,7 +8,9 @@
   `
     <div style="height: 100vh" id="one">one</div>
     <div style="height: 100vh" id="two">two</div>
-    <div style="height: 100vh; width: 200vw" id="three">three</div>`,
+    <div style="height: 100vh; width: 200vw; overflow: auto;" id="three">
+      <div style="height: 300%;">three</div>
+    </div>`,
   async function(browser, accDoc) {
     let onScrolling = waitForEvents([
       [EVENT_SCROLLING, accDoc],
@@ -70,5 +72,25 @@
       scrollEvent3.scrollX > scrollEvent2.scrollX,
       `${scrollEvent3.scrollX} > ${scrollEvent2.scrollX}`
     );
+
+    // non-doc scrolling
+    onScrolling = waitForEvents([
+      [EVENT_SCROLLING, "three"],
+      [EVENT_SCROLLING_END, "three"],
+    ]);
+    await SpecialPowers.spawn(browser, [], () => {
+      content.document.querySelector("#three").scrollTo(0, 10);
+    });
+    let [scrollEvent4, scrollEndEvent4] = await onScrolling;
+    scrollEvent4.QueryInterface(nsIAccessibleScrollingEvent);
+    ok(
+      scrollEvent4.maxScrollY >= scrollEvent4.scrollY,
+      "scrollY is within max"
+    );
+    scrollEndEvent4.QueryInterface(nsIAccessibleScrollingEvent);
+    ok(
+      scrollEndEvent4.maxScrollY >= scrollEndEvent4.scrollY,
+      "scrollY is within max"
+    );
   }
 );
